pipeline_dp/dp_computations.py

```diff
@@ -70,6 +145,32 @@ def _add_random_noise(value: float, eps: float, delta: float,
     raise ValueError("Noise kind must be either Laplace or Gaussian.")
 
 
+def equally_split_budget(eps: float, delta: float, no_mechanisms: int):
+    """Equally splits the budget (eps, delta) between a given number of mechanisms.
+
+    Args:
+        eps, delta: The available budget.
+        no_mechanisms: The number of mechanisms between which we split the budget.
+
+    Raises:
+        ValueError: The number of mechanisms must be a natural non-zero number.
+
+    Returns:
+        An array with the split budgets.
+    """
+    if no_mechanisms <= 0:
+        raise ValueError(
+            "The number of mechanisms must be a natural non-zero number.")
+
+    budgets = [(eps / no_mechanisms, delta / no_mechanisms) for _ in
+               range(no_mechanisms - 1)]
+    last_mechanism_budget = (eps - (no_mechanisms - 1) * (eps / no_mechanisms),
```

Comment:
It's safer to actually sum the budgets and subtract them from the total epsilon and delta, rather than calculating what the sum should be.